Snowflake.Data.Tests/IntegrationTests/SFPutGetTest.cs

```diff
@@ -496,23 +521,51 @@ private void GetFile(DbConnection conn)
                     reader.GetString((int)SFResultSet.PutGetResponseRowTypeInfo.ResultStatus));
 
                 // Check file contents
-                using (var streamReader = new StreamReader(t_outputFilePath))
+                foreach (var line in ReadOutputFileLinesUnzippingIfCompressedBecauseOfAutoCompression())
                 {
-                    while (!streamReader.EndOfStream)
+                    if (string.IsNullOrEmpty(line)) continue;
+                    var values = line.Split(',');
+
+                    for (var i = 0; i < s_colData.Length; i++)
                     {
-                        var line = streamReader.ReadLine();
-                        if (line == null) continue;
-                        var values = line.Split(',');
-
-                        for (var i = 0; i < s_colData.Length; i++)
-                        {
-                            Assert.AreEqual(s_colData[i], values[i]);
-                        }
+                        Assert.AreEqual(s_colData[i], values[i]);
                     }
                 }
             }
         }
 
+        private static string[] ReadOutputFileLinesUnzippingIfCompressedBecauseOfAutoCompression()
```

Comment:
this method basically reads output from a stream, whether it's compressed or not
I would introduce method
DecompressStream
and using a renamed IsCompressedBecauseOfAutoCompression into sth simplier like 'isCompressed'
method reuse it 
draft code:
`using outputStream
      return splitLines( isCompressed ? decompressStream(outputStream) : outputStream);
`

Reply:
I changed it a bit. I would not like to create simply isCompressed method here since it could be a source of confusion. In this test in most cases files just pretend to be compressed.